src/supertux/error_handler.cpp

```diff
@@ -32,9 +54,131 @@ bool ErrorHandler::m_handing_error = false;
 void
 ErrorHandler::set_handlers()
 {
-#ifdef __GLIBC__
   signal(SIGSEGV, handle_error);
   signal(SIGABRT, handle_error);
+}
+
+std::string
+ErrorHandler::get_stacktrace()
+{
+#ifdef WIN32
+  std::stringstream stacktrace;
+
+  // Initialize symbols
+  SymSetOptions(SYMOPT_UNDNAME | SYMOPT_DEFERRED_LOADS);
+  if (!SymInitialize(GetCurrentProcess(), NULL, TRUE))
+  {
+    return "";
+  }
+
+  // Get current stack frame
+  void* stack[100];
+  WORD frames = CaptureStackBackTrace(0, 100, stack, NULL);
+
+  // Get symbols for each frame
+  SYMBOL_INFO* symbol = static_cast<SYMBOL_INFO*>(std::calloc(sizeof(SYMBOL_INFO) + 256 * sizeof(char), 1));
+  symbol->MaxNameLen = 255;
+  symbol->SizeOfStruct = sizeof(SYMBOL_INFO);
+
+  for (int i = 0; i < frames; i++)
+  {
+    SymFromAddr(GetCurrentProcess(), (DWORD64) stack[i], 0, symbol);
+    stacktrace << symbol->Name << " - 0x" << std::hex << symbol->Address << "\n";
+  }
+
+  std::free(symbol);
+  SymCleanup(GetCurrentProcess());
+
+  return stacktrace.str();
+#elif defined(UNIX)
+  void* array[128];
+  size_t size;
+
+  // Get void*'s for all entries on the stack.
+  size = backtrace(array, 127);
+
+  char** functions = backtrace_symbols(array, static_cast<int>(size));
+  if (functions == nullptr)
+    return "";
+
+  std::stringstream stacktrace;
+
+  for (size_t i = 0; i < size; i++)
+    stacktrace << functions[i] << "\n";
+
+  return stacktrace.str();
+#else
+  return "";
+#endif
+}
+
+std::string
+ErrorHandler::get_system_info()
+{
+#ifdef WIN32
+
+  std::stringstream info;
+
+/*
+  // This method reports Windows 8 on my
+  // Windows 10 PC. Disabled.
+  if (IsWindows10OrGreater())
+    info << "Windows 10/11";
+  else if (IsWindows8Point1OrGreater())
+    info << "Windows 8.1";
+  else if (IsWindows8OrGreater())
+    info << "Windows 8";
+  else if (IsWindows7OrGreater())
+    info << "Windows 7";
+  else if (IsWindowsVistaOrGreater())
+    info << "Windows Vista";
+  else if (IsWindowsXPOrGreater())
+    info << "Windows XP";
+  else
+    info << "Windows";
+
+  info << " ";
+*/
```

Comment:
Is this comment needed?

Reply:
not really

Reply:
I think it's good to have, so this can be enabled if someone knows how to fix it.

Reply:
Fair enough.